### DISCUSSION TITLE
Easyengine website with Digital Block Storage

### DISCUSSION THREAD

**Discussion by steve231293** [EasyEngine/easyengine#1803] [Support Requests]
Hello there,
Could you help me to resolve this problem, pls?

I have a WordPress with lot of assets. For eg uploads folder size grow to 200GB. Then i need to use Digital Block Storage service.

I following this ticket to setup:
[](https://www.digitalocean.com/community/questions/how-to-mount-block-storage-volume-to-the-upload-directory-of-my-wordpress-website?)

I copy & symbolink uploads folder successfully from: /mnt/storage/uploads => /opt/easyengine/sites/example.com/wp-content/uploads

But then all images of website's not load properly. 
Maybe Nginx ignore assets out of root folder of website?
Maybe i missing some Nginx config.
Any suggestion ??

Thank you so much.

**Comment by freedog96150** [comment]
I assume that you have this figured out?  Adding some comments here in case anyone stumbles into this with the same question.  You cannot simply move media files in WP.  Their locations have to be updated in the database.  There are many plugins that will accomplish this task for you.  Search for "fix media library" in the WP plugin store.  There are also ways to update media files using the WP-CLI that is installed with EE.  Check out https://developer.wordpress.org/cli/commands/media/  I still recommend the plugin method as it is simple, but I use WP-CLI all the time and it works without adding any new plugins.

I will also comment that while adding block storage is fine, why not add a Digital Ocean space ($5/mo) and use the CDN features of Spaces instead?  You would be able to offload the entire media library from your droplet to the space AND get the benefit of a CDN which helps to increase the speed of the site.  Digital Ocean provides plenty of documentation on adding a space as a CDN to WP sites, including https encryption end-to-end and using a custom domain name such as https://media.mysite.com for SEO.

Again, no wrong answers when storage space is needed, I just like the idea of offloading the library and speeding up the site.

**Comment by steve231293** [comment]
Hello @freedog96150,
Thank you for your reply.

I haven't figured out this problem yet for now.

I tested move uploads folder on localhost machine without any problems.
So i guess i'm missing some configs on Nignx to make it works.

I used DO Block Storage instead of DO Spaces because my website will regenerate thumbnails on fly. For eg: thumbnails will regenerate when someone access specify pages.

Let me know if me going wrong. 

Thank you so much.

**Reply by freedog96150** [reply]
I am not sure that Nginx configs are the problem.  I have moved media libraries, manually uploaded via FTP, moved to CDN's etc and never had to change an NGX config.

Let's take a step back and make sure that your mounted directories have the correct owner and file permissions.  The uploads folder should be owned by the web user - www-data in this case.  The file permissions should all be set to 644 for files and 755 for directories. Even with a symlinked folder, you want to make sure the actual folders are set properly and match the owner and permissions of the symlink.  

Once you have verified that the owner and file permissions are correct, navigate to your /opt/easyengine/sites/example.com/wp-content/uploads folder and make sure you can list the contents of a few directories.  You should see your files.  If that is the case then you are back to my original comment which is to use a plugin to fix the database links for the media library.  Remember that to show in the media library the file must be physically present in the proper folder and must have an entry in the database.  Simply copying a file into a folder does not guarantee that the media library "sees" the file.

And if ownership and permissions are not correct, the web server cannot display the file properly and you'll have all sorts of problems managing the file via the media library.
